tasks/managed/push-rpm-to-koji/push-rpm-to-koji.yaml

```diff
@@ -140,7 +140,26 @@ spec:
               draft_flag="--draft"
           fi
 
-          printf "Import rpm %s with tags $KOJI_TAGS ...\n" "$PACKAGE_NVR"
+          # Get Koji build target from the RPM image annotation to tag the
+          # imported build later.
+          koji_target=$(
+            oras manifest fetch "$containerImage" |
+              jq --exit-status -r '.annotations."koji.build-target"'
+          )
+          if [ -z "$koji_target" ]; then
+              echo "ERROR: No Koji build target found in the container image annotations."
+              exit 1
+          fi
+          # For draft builds, remove "-candidate" tag suffix if present and
+          # append "-draft".
+          if [[ "$draft" == "true" && "$koji_target" != *-draft ]]; then
+              koji_tag=${koji_target%-candidate}-draft
+          else
+              koji_tag=${koji_target}
+          fi
+          KOJI_TAGS="$koji_tag $KOJI_TAGS"
+
+          printf "Import rpm %s with tags %s ...\n" "$PACKAGE_NVR" "$KOJI_TAGS"
```

Comment:
Would it make sense to add or modify tests to cover this new logic?

Reply:
I've added some tests.

Unfortunately, I fail to run the tests locally with the CI script `.github/scripts/test_tekton_tasks.sh`. For some reason the workspace data does not seem to persist between tasks - `run-task` does not see `/workspace/data/snapshot_spec.json`.